dracut/30afterburn/afterburn-hostname.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Afterburn Hostname
+# Azure needs to fetch the hostname in the initramfs
+ConditionKernelCommandLine=|ignition.platform.id=azure
+ConditionFirstBoot=yes
+After=ignition-disks.service
```

Comment:
@yuqi-zhang did you test this service unit on an actual Azure machine? Because I think I made a mistake and this should be `After=ignition-mount.service`.

Reply:
I guess since it only writes to `/sysroot/etc`, technically it just needs `After=initrd-root-fs.target`. That'll work on both FCOS and RHCOS (which doesn't have `ignition-mount.service` yet).

Reply:
I added `After=initrd-root-fs.target`
`Wants=initrd-root-fs.target` based on https://github.com/coreos/ignition-dracut/blob/spec2x/dracut/30ignition/coreos-static-network.service

Reply:
Ack. And do we care about `ignition-remount-sysroot.service` here? (I'm not so familiar with the new Ignition flow in initramfs)

Reply:
I suppose for safety we should? I'm also not sure on which systems that one actually runs.
And to your earlier point, I didn't test on Azure since I didn't have creds at the time, I made a mock unit with the same dependency tree to run on metal, and it worked as expected.

Reply:
Added